omniduct/databases/base.py

```diff
@@ -258,22 +240,43 @@ def render_template(self, name_or_statement, context=None, by_name=False):
         else:
             statement = name_or_statement
 
-        if not context:
-            context = {}
-
-        # Substitute in any other named statements recursively
-        while '{{{' in statement or '{{%' in statement:
-            statement = Template(statement,
-                                 block_start_string='{{%',
-                                 block_end_string='%}}',
-                                 variable_start_string='{{{',
-                                 variable_end_string='}}}',
-                                 comment_start_string='{{#',
-                                 comment_end_string='#}}',
-                                 undefined=StrictUndefined).render(getattr(self, '_templates', {}))
-
-        # Evaluate final template in provided context
-        statement = Template(statement, undefined=StrictUndefined).render(context)
+        try:
+            from sqlalchemy.sql.base import Executable
+            if isinstance(statement, Executable):
+                statement = str(statement.compile(compile_kwargs={"literal_binds": True}))
+        except ImportError:
+            pass
+
+        # If templating enabled, render template
+        if template:
+
+            if context is None:
```

Comment:
I had `if not context` to be nice to people who do things like: `context=False`. Perhaps that's not necessary.